### PR TITLE
Fix/map 126 refactor comments functionality

### PR DESCRIPTION
**Pull request by dobbi-free**
Depends on the presence of a field in store userAuth.id

### CODE REVIEW

File: backend/models/CommentModel.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  text: {
+    type: String,
+    required: true
+  },
+  likes: {
+    type: Array
+  },
+  dislikes: {
+    type: Array
+  },
```
Comment (6mahdapihka9) [inline backend/models/CommentModel.ts]: I think this fields should have default values as [ ] for each

File: frontend/src/components/BigPopup/CommentSection/Comment.tsx [hunk 2]
```diff
@@ -1,22 +1,24 @@
...
 interface Props {
   text: string;
   createdAt: Date;
+  displayName: string;
+  imageUrl: string;
 }
 
-const Comment = ({ text, createdAt }: Props) => {
+const Comment = ({ text, createdAt, displayName, imageUrl }: Props) => {
   const d = new Date(createdAt);
```
Comment (6mahdapihka9) [inline frontend/src/components/BigPopup/CommentSection/Comment.tsx]: I recommend to name props as authorsName and authorsImage or authorsAvatar

File: frontend/src/components/BigPopup/CommentSection/CommentForm.tsx [hunk 3]
```diff
@@ -15,16 +15,17 @@ const CommentForm = () => {
   const { _id: locationId } = useTypedSelector(state => state.popupLocation);
 
   const onSendComment = () => {
-    const commentBody: Comment = {
+    const commentBody: Comment<string> = {
       author: userId,
+      locationId: locationId || '',
       text: commentText,
```
Comment (6mahdapihka9) [inline frontend/src/components/BigPopup/CommentSection/CommentForm.tsx]: how is it possible to send an empty string as a location id?? 🤔

File: frontend/src/redux/ts-types/popupLocation.tsx [hunk 4]
```diff
@@ -1,12 +1,19 @@
-export type Comment = {
-  author: string;
+export type Comment<T> = {
+  _id?: string;
+  author: T;
+  locationId: string;
   text: string;
```
Comment (6mahdapihka9) [inline frontend/src/redux/ts-types/popupLocation.tsx]: nice flex :)

File: frontend/src/components/BigPopup/CommentSection/CommentForm.tsx (hunk 3 above)
Comment (6mahdapihka9) [reply frontend/src/components/BigPopup/CommentSection/CommentForm.tsx]: need explanation

**Review by 6mahdapihka9 — CHANGES_REQUESTED** [review]
seem good besides one thing, why did you separate comments from locations on the backend but not on the frontend?

**Review by Yojuks — APPROVED** [review]

File: frontend/src/components/BigPopup/CommentSection/Comment.tsx [hunk 5]
```diff
@@ -1,22 +1,24 @@
...
 interface Props {
   text: string;
   createdAt: Date;
+  authorsName: string;
+  authorsImage: string;
 }
 
-const Comment = ({ text, createdAt }: Props) => {
+const Comment = ({ text, createdAt, authorsName, authorsImage }: Props) => {
   const d = new Date(createdAt);
```
Comment (obozhko-dev) [inline frontend/src/components/BigPopup/CommentSection/Comment.tsx:22]: To format date you can use something like this d.toLocaleDateString('en-US') , if you want, not critical . About format date https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

**Review by obozhko-dev — COMMENTED** [review]
1. Contact with Oleksandr Kulesh and confirm with his redux structure , pls. He working with refactoring redux now 
2.  In comment list : 
2.1 If I add short comment - it not centered, may be it will be in center of box vertically
2.2. Big comment breaks the date
2.3. Likes don`t work, may be it another task ...
image

3. maybe you can add functional for sorting comment for: date up/down, like count up/down may be it will be another task ...

File: frontend/src/components/BigPopup/CommentSection/Comment.tsx (hunk 5 above)
Comment (dobbi-free) [reply frontend/src/components/BigPopup/CommentSection/Comment.tsx:22]: What format do you suggest? Show the exact example that you think is best. 
I know about styling issues but layout wasn't my task. It should be improve in future

**Review by 6mahdapihka9 — APPROVED** [review]

**Pull request merged**